core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/base/NetworkV4Base.java

```diff
@@ -58,6 +60,16 @@ public void setSubnetCIDR(String subnetCIDR) {
         this.subnetCIDR = subnetCIDR;
     }
 
+    public Optional<Boolean> isNoPublicIp() {
+        if (azure != null) {
+            return Optional.of(azure.getNoPublicIp());
+        } else if (gcp != null) {
+            return Optional.of(gcp.getNoPublicIp());
```

Comment:
`Optional.ofNullable` would be more secure against nullpointer

Reply:
Ah, thank you! I was expecting the behavior of `ofNullable`.

Reply:
you know that this means you have to cover this case with unit test :)

Reply:
Lol, thanks for keeping me honest. Please see `NetworkV4BaseTest`.